code/game/objects/structures/training_machine.dm

```diff
@@ -140,7 +142,7 @@
 			last_attack_time = world.time
 			attack_cooldown = rand(MIN_ATTACK_DELAY,MAX_ATTACK_DELAY)
 
-	addtimer(CALLBACK(src, .proc/do_movement), MAX_SPEED - move_speed)
+	addtimer(CALLBACK(src, .proc/do_movement), abs(MAX_SPEED - move_speed) + 1) // We want to ensure this is never <=0
```

Comment:
Are you sure this is what you want and not `max`? As in, `max(MAX_SPEED - move_speed, 1)`?

Reply:
max works better for this u right